history.go

```diff
@@ -7,37 +7,60 @@ import (
 )
 
 var (
-	ErrBadEpoch    = errors.New("[merkletree] Next epoch's STR has bad epoch")
-	ErrSTRNotFound = errors.New("[merkletree] STR not found")
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
 )
 
+type Epoch int64
+
 type History struct {
-	str           *SignedTreeRoot
-	epochInterval int64
+	currentSTR    *SignedTreeRoot
+	hashChain     map[Epoch]*SignedTreeRoot
+	epochs        []Epoch // slice of epochs in hashChain
+	epochInterval Epoch
```

Comment:
> Does the ``MerkleTree`` need policies? Or should that belong to ``History``?

We put the policy in the STR (not the tree) for two reasons. Many policies are tree-specific (e.g. the key for computing the VRF), and because we want the server to commit (not cryptographically) to the policies by including them in the STR.

From the paper, section 3.3: "P may include, for example, the key K_VRF used to generate private indices, an expected time the next epoch will be published, as well as the cryptographic algorithms in use, protocol version numbers, and so forth."

I think this also leads us to answer @c633's question 
> Is the policies immutable? Or could the developers change it at some epochs?

No, the policies aren't immutable, the developers can change them between epochs. Protocol version numbers are a great example of something that could change. The server could also change it's VRF public key between epochs (this helps re-randomize the order of leaves), or change its epoch interval.